### ISSUE TITLE
Sharing cookies across different WebViews

### ISSUE THREAD

**Issue by Stefanama** [react-native-community/react-native-webview#1350]
Hello 👋
in our app we use a WebView to log the user in and then we use the rn app normally by capturing the access token.
In some places we rely on WebViews to access authenticated pages.
On android it looks like the cookies are shared across different WebViews so the pages will work fine, on iOS however the user gets redirected to the login page within that same WebView, so I'm guessing the cookies aren't shared. 
I've tried setting `sharedCookiesEnabled` on the WebView but doesn't seem to work.

Anyone have any ideas? Has run into similar situations?

Thanks 😊

**Comment by kyorkston** [comment]
Our first iteration of this same problem was to use the prop: [`injectedJavascript`](https://github.com/react-native-community/react-native-webview/blob/master/docs/Reference.md#injectedjavascript), where inside you can use `ReactNativeWebview.postMessage(document.cookie)`to pass the whole cookie string listened to by the [`onMessage`](https://github.com/react-native-community/react-native-webview/blob/master/docs/Reference.md#onmessage) prop on the WebView component.
```
injectedJavascript=";(function() {
                        ReactNativeWebView.postMessage(document.cookie);
            })()
            "
```
It passes on an event ([`WebViewMessageEvent/WebViewMessage`](https://github.com/react-native-community/react-native-webview/blob/master/src/WebViewTypes.ts#L122)) where you can access the data from the posted message.

Also depending on the routing of the app inside the webview there is [`onNavigationStateChange`](https://github.com/react-native-community/react-native-webview/blob/master/docs/Reference.md#onnavigationstatechange), which listens to when the webview's navigation starts and ends so you could intercept that to grab the cookie or close the webview, etc. 
I hope this helps! It was a lot of back and forth research when we were implementing login via a webview 🙂

**Comment by mirsylvain** [comment]
I ran into the exact situation.
I used an authentication system from a first Webview which set Cookies which are mandatory to access to others ressources I load with others Webview but the cookies (altought they are on the same domain) are not transimtted.

I displayed with the CookieManager the content of the cookie storage and there are effectively set.

I set the property sharedCookiesEnabled on each Webview but this doen't seem to have any effect.
